src/plugins/data_source_management/public/components/data_source_selector/data_source_selector.tsx

```diff
@@ -76,6 +79,20 @@ export class DataSourceSelector extends React.Component<
             allDataSources: fetchedDataSources,
           });
         }
+        if (this.props.showDefault) {
+          const dataSources = getFilterDataSources(
+            this.state.allDataSources,
+            this.props.dataSourceFilter
+          );
+          const defaultDataSource = getDefaultDataSource(
+            dataSources,
+            LocalCluster,
+            this.props.uiSettings,
+            this.props.hideLocalCluster,
+            this.props.defaultOption
+          );
+          this.props.showDefault(defaultDataSource);
```

Comment:
can we reuse the existing callback function? onSelectedDataSource

Reply:
I can make the change. Just worry about if it would break any existing behavior. Let me change on this